WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -660,6 +725,62 @@ class PostListViewModel @Inject constructor(
         }
     }
 
+    private fun loadPostFromRemote(localPostId: Int) {
+        // We need network connection to load a remote post
+        if (!checkNetworkConnection()) {
+            return
+        }
+
+        val post = postStore.getPostByLocalPostId(localPostId)
+        if (post != null) {
+            originalPostCopyForConflictUndo = post.clone()
+            dispatcher.dispatch(PostActionBuilder.newFetchPostAction(RemotePostPayload(post, site)))
+            _toastMessage.postValue(ToastMessageHolder(R.string.toast_conflict_updating_post, Duration.SHORT))
+        } else {
+            localPostIdForConflictResolutionDialog = null
+        }
+    }
+
+    private fun savePostFromLocal(localPostId: Int) {
+        // We need network connection to push local version to remote
+        if (!checkNetworkConnection()) {
+            return
+        }
+
+        val post = postStore.getPostByLocalPostId(localPostId)
+        if (post == null) {
+            return
+        }
+
+        localPostIdForConflictResolutionDialog = null
+        // keep a copy for undoing
+        originalPostCopyForConflictUndo = post.clone()
+
+        // we make remoteLastModified match local lastModified to clear the "conflicted" flag in the Post list only
+        post.remoteLastModified = post.lastModified
+        dispatcher.dispatch(PostActionBuilder.newUpdatePostAction(post))
+
+        // and now show a snackbar, acting as if the Post was pushed, but effectively push it after the snackbar is gone
+        var isUndoed = false
```

Comment:
Nitpick: I think it'd be better to check `originalPostCopyForConflictUndo == null` instead of keeping another variable for this. We need that information to update the post anyway, so it'd serve as a double check.

Reply:
I think in this case it's clearer with the explicit flag `isUndoed`, otherwise the implicit check for `originalPostCopyForConflictUndo == null` doesn't immediately uncover there's a relationship in the order of execution between the `undoAction` callback and the `onDismissAction` callback. I've tested both and it works but I prefer to have this dedicated variable to make it clear that the code in `onDismissAction  ` is only to be executed when the user tapped on `undo`, rather on whenever it runs and the check `originalPostCopyForConflictUndo == null` happens to be true.

Reply:
Sure, that works for me. We might still need to check for `originalPostCopyForConflictUndo == null`, but the extra flag doesn't hurt and might make it more obvious as you suggested 👍 

I am going to check whether we really need to null check `originalPostCopyForConflictUndo` during my next testing.

Reply:
I think we are all good now, resolving this issue.